app.js

```diff
@@ -0,0 +1,132 @@
+// //1 object with array property
+// let myObject = {
+//     thisKey: [1,2,3,4]
+// };
+// console.log(myObject.thisKey[0]);
+//
+// //2 object with inner object
+// let obj = {
+//     good: {
+//         bad: "great"
+//     }
+// };
+// console.log(obj.good.bad);
+//
+// // 3
+// let otherObj = {
+//     doSomething: function (){
+//         console.log("this is a function");
+//     }
+// };
+// otherObj.doSomething();
+//
+// //4 array with object
+// let myArray = [
+//     {cool: "stuff"}
+// ];
+// console.log(myArray[0]);
+//
+// //5 array with array
+// let otherArr = [
+//     [1,2,3]
+// ];
+// console.log(otherArr[0][1]);
+//
+// // 6
+// let arr = [1,2, function(){console.log("this is an array function")}];
+// arr[2]();
+//
+// //7
+// let objArr = {
+//     [1,2,3,4]
+// };
+// for(i=0; i < objArr.length; i++){
+//     console.log(objArr[0])
+// }
+//
+// //8 moving on
+// let array1 = [["a", "b", "c"]];
+// for(i=0; i < array1.length; i++){}
+
+/* create callback */
+//
+// let abc = function (){
+//     console.log("hi");
+//     param();
+// };
+// let zyx = function (param){
+//     console.log("bye");
+// };
+// abc(zyx());
```

Comment:
close but not quite.  it causes an error. see corrections below.
1) abc takes the param not xyc 
2) don't call zyx, just pass it 
```
let abc = function (param){
    console.log("hi");
    param();
};
let zyx = function (){
    console.log("bye");
};
abc(zyx);
```

Reply:
-0.5